front/pages/edit.vue

```diff
@@ -230,45 +219,72 @@ function initializeFlagResolution() {
+
+  if (hasChanged) {
+    initBufferCanvas();
+  }
+
   return hasChanged;
 }
 
-//Draw EVERY PIXEL of the map given
-function drawFlag(pixelMap) {
-  canvasDrawingContext.fillRect(cameraPositionX, cameraPositionY, canvas.width*cameraZoom, canvas.height*cameraZoom);
-  for (let i = 0; i < pixelMap.length; i++) {
-    for (let j = 0; j < pixelMap[i].length; j++) {
-      drawPixel(i, j, pixelMap[i][j]);
-    }
+function updateFlagPixelMap(data) {
+  flagPixelMap = mapFlagDataToWorldCoordinates(data, flagIndexToCoordinateCache);
+  drawFlagToBuffer();
+}
+
+function drawFlagToBuffer() {
+  const ctx = bufferCanvas.getContext('2d');
+
+  clearCanvas(bufferCanvas);
+
+  const imageData = ctx.createImageData(flagWidth, flagHeight);
+  const textureData = makeFlagTextureArray(flagWidth, flagHeight, flagPixelMap);
+  for(let i = 0; i < textureData.length; i++) {
+    imageData.data[i] = textureData[i] || 0;
+  }
+  ctx.putImageData(imageData, 0, 0);
+}
+
+function drawPixelToBuffer(x, y, hexColor) {
+  const ctx = bufferCanvas.getContext('2d');
+  ctx.fillStyle = hexColor;
+  ctx.fillRect(x, y, 1, 1);
+}
+
+function clearCanvas(canvas) {
+  const ctx = canvas.getContext('2d');
+
+  ctx.save();
+  ctx.setTransform(1, 0, 0, 1, 0, 0);
+  ctx.clearRect(0, 0, canvas.width, canvas.height);
+  ctx.restore();
+}
+
+function drawWorld() {
+  if (!canvas) {
+    console.warn('drawWorld: canvas not ready');
+    return;
   }
+
+  clearCanvas(canvas);
+
+  const ctx = canvas.getContext('2d');
+  ctx.imageSmoothingEnabled = false;
+  ctx.drawImage(bufferCanvas, 0, 0, canvas.width, canvas.height);
 }
 
 //Draw an overlay to find the user pixel on the whole flag
-function drawOverlay() {
-  for (let i = 0; i < flagPixelMap.length; i++) {
-    for (let j = 0; j < flagPixelMap[0].length; j++) {
-      if (!(i == userXPixel && j == userYPixel)) {
-        drawPixel(i, j, { hexColor : "#090909e0" });
+function drawOverlayToBuffer() {
+  for (let x = 0; x < flagPixelMap.length; x++) {
+    for (let y = 0; y < flagPixelMap[0].length; y++) {
+      if (!(x == userXPixel && y == userYPixel)) {
+        drawPixelToBuffer(x, y, "#090909e0");
       } else {
-        drawPixel(i, j, { hexColor : "#00ff16" });
+        drawPixelToBuffer(x, y, "#00ff16");
       }
       }
     }
-}
-
-//Draw a pixel on a coord given (x,y,clr), if changetexture is set to true, change the value on the map
-//You can change the size and the context to draw, default is flag context
-function drawPixel(x, y, pixel, changeTexture = false, size = 1, ctx = canvasDrawingContext) {
-  if (ctx && pixel?.hexColor) {
-    let drawWidth = (canvas.width / flagWidth) * size;
-    let drawHeight = (canvas.height / flagHeight) * size;
-    ctx.fillStyle = pixel.hexColor;
-    ctx.fillRect(x * drawWidth, y * drawHeight, drawWidth, drawHeight);
-    if (changeTexture) {
-      flagPixelMap[x][y].hexColor = pixel.hexColor;
-    }
-    ctx.fillStyle = "#ffffff";
-  }
+  drawWorld();
```

Comment:
The function is called `drawOverlayToBuffer` but once this is done, there is a call to `drawWorld`